core/commands/add.go

```diff
@@ -227,9 +250,24 @@ func add(n *core.IpfsNode, reader io.Reader) (*dag.Node, error) {
 	return node, nil
 }
 
-func addFile(n *core.IpfsNode, file files.File, out chan interface{}, progress bool, wrap bool) (*dag.Node, error) {
+func addFile(n *core.IpfsNode, file files.File, out chan interface{}, progress bool, wrap bool, hidden bool, ignoreFilePatterns []ignore.GitIgnore) (*dag.Node, error) {
```

Comment:
On Wed, May 06, 2015 at 10:25:36PM -0700, Juan Batiz-Benet wrote:
> > -func addFile(n *core.IpfsNode, file files.File, out chan interface{}, progress bool, wrap bool) (*dag.Node, error) {
> > +func addFile(n *core.IpfsNode, file files.File, out chan interface{}, progress bool, wrap bool, hidden bool, ignoreFilePatterns []ignore.GitIgnore) (*dag.Node, error) {
> 
> this function is getting crazy. it may be useful to construct an
> `adder` object that holds the configuration:

Can we have a public function that uses a channel to support the
progress bar [1]?  I think we should be thinking over the adder UI
very carefully before altering a public API or creating functionality
that's *not* part of a public UI.  But I haven't looked through the
rest of this PR yet.  I'll try and do that tomorrow, but don't wait
for my review if I'm too slow ;).

[1]: https://github.com/ipfs/go-ipfs/pull/1127#issuecomment-96024500